### ISSUE TITLE
Avoid masking return values when using `< <(command)`

### ISSUE THREAD

**Issue by felipecrs** [koalaman/shellcheck#2784]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# shellcheck enable=check-extra-masked-returns

readarray -t arr < <(false)

cat < <(false)
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Maybe SC2312 (check-extra-masked-returns) should be extended to cover these cases (which is by itself similar to [SC2155](https://www.shellcheck.net/wiki/SC2155)):

Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore). Because, otherwise that's exactly what happens:

```console
$ false; echo $?
1

$ readarray -t arr < <(false); echo $?
0

$ cat < <(false)
0
```

**Comment by felipecrs** [comment]
Nevermind, that's exactly what's happening already.
